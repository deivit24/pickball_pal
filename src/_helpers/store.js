import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
