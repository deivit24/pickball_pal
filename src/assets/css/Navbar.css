button {
  outline: none !important;
}

a {
  color: inherit !important;
  outline: none !important;
  text-decoration: none !important;
}
