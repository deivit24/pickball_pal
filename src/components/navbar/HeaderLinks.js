import React from 'react';
// react components for routing our app without refresh
import { Link } from 'react-router-dom';
//logout action redux
import { logout } from '../../_actions/auth';
import { useSelector, useDispatch } from 'react-redux';
// @material-ui/core components
import { makeStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';

// @material-ui/icons

// core components

import Button from './Button';
import styles from '../../assets/js/headerLinksStyle';

const useStyles = makeStyles(styles);

export default function HeaderLinks(props) {
  const auth = useSelector((store) => store.auth);
  const dispatch = useDispatch();
  const classes = useStyles();

  const { isAuth, loading } = auth;
  const handleLogout = async (e) => {
    e.preventDefault();
    dispatch(logout());
  };
  const authLinks = (
    <>
      <ListItem className={classes.listItem}>
        <Link to="/dashboard">
          <Button color="transparent" className={classes.navLink}>
            <i className="fas fa-grip-horizontal"></i> Dashboard
          </Button>
        </Link>
      </ListItem>
      <ListItem className={classes.listItem}>
        <Link to="/posts">
          <Button color="transparent" className={classes.navLink}>
            <i className="fas fa-comment-alt"></i> Posts
          </Button>
        </Link>
      </ListItem>
      <ListItem className={classes.listItem}>
        <Button
          onClick={handleLogout}
          color="transparent"
          className={classes.navLink}
        >
          <i className="fas fa-sign-out-alt"></i> Logout
        </Button>
      </ListItem>
    </>
  );

  const guestLinks = (
    <>
      <ListItem className={classes.listItem}>
        <Link to="/login">
          <Button color="transparent" className={classes.navLink}>
            Login
          </Button>
        </Link>
      </ListItem>
      <ListItem className={classes.listItem}>
        <Link to="/register">
          <Button color="transparent" className={classes.navLink}>
            Register
          </Button>
        </Link>
      </ListItem>
    </>
  );
  return (
    <List className={classes.list}>
      <ListItem className={classes.listItem}>
        <Link to="/profiles">
          <Button color="transparent" className={classes.navLink}>
            <i className="fas fa-users"></i> Players
          </Button>
        </Link>
      </ListItem>
      {!loading && <> {isAuth ? authLinks : guestLinks} </>}
    </List>
  );
}
