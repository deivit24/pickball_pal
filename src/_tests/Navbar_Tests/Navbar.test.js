import React from 'react';
import { render } from '@testing-library/react';
import Navbar from '../../components/navbar/Navbar';
import { MemoryRouter } from 'react-router';
import Logo from '../../assets/img/logo.png';
import HeaderLinks from '../../components/navbar/HeaderLinks';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing Navbar

describe('Testing the Navbar', () => {
  const color = 'white';

  const changeColorOnScroll = {
    height: 100,
    color: 'white',
  };

  const handleSubmit = () => {
    console.log('Testing Handle Submit');
  };
  it('it renders Navbar without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <Navbar
            color={color}
            brand={Logo}
            rightLinks={<HeaderLinks />}
            changeColorOnScroll={changeColorOnScroll}
          />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <Navbar
            color={color}
            brand={Logo}
            rightLinks={<HeaderLinks />}
            changeColorOnScroll={changeColorOnScroll}
          />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
