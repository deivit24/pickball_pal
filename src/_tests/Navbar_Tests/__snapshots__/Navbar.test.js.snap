// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Navbar matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-67 makeStyles-white-82 mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-container-70 MuiToolbar-gutters"
    >
      <div
        class="makeStyles-flex-71"
      >
        <a
          href="/"
        >
          <img
            alt=""
            id="Logo"
            src="logo.png"
          />
        </a>
      </div>
      <div
        class="PrivateHiddenCss-smDown-89"
      >
        <ul
          class="MuiList-root makeStyles-list-99 MuiList-padding"
        >
          <li
            class="MuiListItem-root makeStyles-listItem-100 MuiListItem-gutters"
          >
            <a
              href="/profiles"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-111 makeStyles-transparent-125 makeStyles-navLink-102"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <i
                    class="fas fa-users"
                  />
                   Players
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
